celer/homotopy.py

```diff
@@ -144,16 +144,19 @@ def celer_path(X, y, pb, eps=1e-3, n_alphas=100, alphas=None,
         (Is returned only when ``return_thetas`` is set to True).
     """
 
-    if pb.lower() not in ("lasso", "logreg", "grouplasso"):
+    pb = pb.lower()
+    if pb not in ("lasso", "logreg", "grouplasso"):
         raise ValueError("Unsupported problem %s" % pb)
-    if pb.lower() == "lasso":
+    # prevent referenced before assigned error
+    grp_ptr, grp_indices, n_groups = None, None, None
```

Comment:
can you avoid diff like that ? They make the PR harder to review. Can you revert ?

Reply:
``grp_ptr, grp_indices, n_groups = None, None, None`` is necessary to avoid breaking down``_check_weights``

Reply:
`grp_ptr, grp_indices` are not needed in `check_weights`, so we can remove these 2

Fewer variable definitions makes simpler code, fewer sources of mistake.

Can you think of a design where you would'nt have to do this ?

Reply:
I added these two variables for code readability.
I will remove them and keep ``n_groups``